Support TextField multi-line hint text #20941

trevorwang opened the pull request.

trevorwang committed:
Support TextField multi-line hint text #20941

trevorwang commented:
<img width="333" alt="screen shot 2018-12-05 at 3 44 44 pm" src="https://user-images.githubusercontent.com/121966/49498042-d532c780-f8a4-11e8-807f-4f8089d78997.png">

xster commented:
This is probably for @willlarche since the Material spec seems to say single line helper text only.

willlarche commented:
This is actually a supported feature for text fields we are working on propogating to all platforms. @trevorwang just saved us some time!

willlarche reviewed (approved):
Please add a test for the case of more than 1 line of text and make one of the text fields in the example in flutter_gallery show 2 lines.

trevorwang committed:
Add test case and demo for multiple line hint text

trevorwang commented:
@willlarche I have added the test case and updated the demo in flutter_gallery.

xster reviewed (commented):
LGTM. Thanks for your contribution

trevorwang commented on examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart:
```diff
@@ -233,7 +233,7 @@ class TextFormFieldDemoState extends State<TextFormFieldDemo> {
                   decoration: const InputDecoration(
                     border: OutlineInputBorder(),
-                    hintText: 'Tell us about yourself',
+                    hintText: 'Tell us about yourself (e.g., write down what you do or what hobbies you have)',
```
yes.  It will use the maxLines value from TextField. Here's the screenshot.
![screen shot 2018-12-12 at 10 40 34 am](https://user-images.githubusercontent.com/121966/49843520-7fa57000-fdfa-11e8-97f3-c38c0850210b.png)

xster commented on packages/flutter/lib/src/material/input_decorator.dart:
```diff
@@ -2081,6 +2084,15 @@ class InputDecoration {
 
+  /// The maximum number of lines the [hintText] can occupy.
+  ///
+  /// Defaults to null, which means that the [hintText] will be limited
```
We should describe that cascading behavior in the doc here in that case.

trevorwang committed:
Update comments about hintMaxLines

Set hintMaxLines with TextField's maxLines attribute only if it
is not provided by InputDecoration

trevorwang committed:
Remove spaces
